core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCosmicData.java

```diff
@@ -65,18 +67,34 @@ public void importData() throws IOException, DaoException {
                 }
                 
                 String id = parts[2];
-                if (!id.matches("COSM[0-9]+")) {
+                if (!id.matches("COS(M|V)[0-9]+")) {
                     System.err.println("Wrong cosmic ID: "+id);
                 } else {
                     id = id.substring(4);
                 }
                 
-                Matcher m = p.matcher(parts[7]);
+                String infoColumnValue = parts[7];
+                Matcher m = p.matcher(infoColumnValue);
                 if (m.find()) {
-                    String gene = m.group(1);
-//                    if (gene.contains("_ENST")) {
-//                        gene = gene.substring(0,gene.indexOf("_ENST"));
-//                    }
+                    // Remove fields that won't be used, so that they don't interfere with column indices
+                    // If field in last place, it won't be removed, but it won't interfere either
+                    infoColumnValue = infoColumnValue.replaceAll("LEGACY_ID=[^;]+;","")
+                                                        .replaceAll("HGVSP=[^;]+;","")
+                                                        .replaceAll("HGVSG=[^;]+;","")
+                                                        .replaceAll("HGVSC=[^;]+;","")
+                                                        .replaceAll("SNP;","");
```

Comment:
What is the reason for removing these fields?

Reply:
It is common that in the same file the fields HGVSP, HGVSG, HGVSC and SNP are present only in some rows. This affects the selection after the first row.

Reply:
But the presence of these fields does not interfere with the function of the script, does it? At least I do not see how that is a problem.

Reply:
It does when the column 7 looks like:
```
GENE=; STRAND=; CDS=; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSPC; HGVSP=; AA=; CNT=
GENE=; STRAND=; CDS=; HGVSP=; HGVSP=; SNP; AA=; CNT=
```
In this case only rows 1 and 2 would be correct.